File: assignments/20-3-16/이한범/mergeList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* mergedList = NULL;
        ListNode* first = NULL;
        
        while(l1 && l2){
            if(l1->val < l2->val) {
                if(first == NULL) {
                    first = l1;
                    mergedList = l1;
                    l1 = l1->next;
                    continue;
                }
                mergedList->next = l1;
                l1 = l1->next;
                mergedList = mergedList->next;
            }
            else {
                if(first == NULL) {
                    first = l2;
                    mergedList = l2;
                    l2 = l2->next;
                    continue;
                }
                mergedList->next = l2;
                l2 = l2->next;
                mergedList = mergedList->next;
            }
        }
        if(l1) {
            if(first) mergedList->next = l1;
            else first = l1;
        }
        else {
            if(first) mergedList->next = l2;
            else first = l2;
        }
        
        return first;
    }
};

File: assignments/20-3-16/차봉건/MergeList.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:

        if l1 == None and l2 == None:
            return None
        elif l1 == None:
            return l2        
        elif l2 == None:
            return l1
        
        if(l1.val >= l2.val):
            merged = ListNode(l2.val)
            l2 = l2.next
        else:
            merged = ListNode(l1.val)
            l1 = l1.next
            
        tmp_node = merged
        while l1 != None and l2 != None:
            if l1.val >= l2.val:
                new_node = ListNode(l2.val)
                tmp_node.next = new_node
                tmp_node = tmp_node.next
                l2 = l2.next
            else:
                new_node = ListNode(l1.val)
                tmp_node.next = new_node
                tmp_node = tmp_node.next
                l1 = l1.next
        
        while l1 != None:
            new_node = ListNode(l1.val)
            tmp_node.next = new_node
            tmp_node = tmp_node.next
            l1 = l1.next
        while l2 != None:
            new_node = ListNode(l2.val)
            tmp_node.next = new_node
            tmp_node = tmp_node.next
            l2 = l2.next
            
        return merged
        
